Doc/library/weakref.rst

```diff
@@ -167,6 +167,16 @@ See :ref:`__slots__ documentation <slots>` for details.
    application without adding attributes to those objects.  This can be especially
    useful with objects that override attribute accesses.
 
+   Note that when a key with equal value to an existing key (but not equal identity)
+   is inserted into the dictionary, it replaces the value but does not replace the
+   existing key. Due to this, when the reference to the original key is deleted, it
+   also deletes the entry in the dictionary::
+
+      d = WeakKeyDictionary()
```

Comment:
Can you add a definition of `k1` and `k2`  here? Without it, this example is hard to understand.
```suggestion
      class T(str):
          pass
      k1 = T('hello')
      k2 = T('hello')
      d = WeakKeyDictionary()
```